tests/templates/test_integration.py

```diff
@@ -647,3 +647,82 @@ def circuit():
+        nondiffable.update(diffable)
+
+        # construct qnode with consecutive wires
+        dev_consec = qml.device('default.qubit', wires=n_wires)
+
+        @qml.qnode(dev_consec)
+        def circuit_consec():
+            template(**nondiffable)
+            return qml.expval(qml.Identity(wires=0))
+
+        # construct qnode with nonconsecutive wires
+        nonconsecutive_wires = np.random.choice(['a', 'b', 'c', 'd', 'e', 'f'], size=n_wires, replace=False)
+        kwargs_nonconsec = nondiffable.copy()
+        if 'wires' in kwargs_nonconsec:
+            kwargs_nonconsec['wires'] = nonconsecutive_wires
+        # DoubleExcitationLayers does not have a wires kwarg
+        if 'wires1' in kwargs_nonconsec:
+            kwargs_nonconsec['wires1'] = nonconsecutive_wires[:2]
+            kwargs_nonconsec['wires2'] = nonconsecutive_wires[2:]
+        # some kwargs in UCSSD need to be manually replaced
+        if 'ph' in kwargs_nonconsec:
+             kwargs_nonconsec['ph'] = [nonconsecutive_wires[:3], nonconsecutive_wires[1:]]
+        if 'pphh' in kwargs_nonconsec:
+             kwargs_nonconsec['pphh'] = [[nonconsecutive_wires[:2], nonconsecutive_wires[2:]]]
+
+        dev_nonconsec = qml.device('default.qubit', wires=nonconsecutive_wires)
+
+        @qml.qnode(dev_nonconsec)
+        def circuit_nonconsec():
+            template(**kwargs_nonconsec)
+            return qml.expval(qml.Identity(wires=nonconsecutive_wires[0]))
+
+        # run circuits
+        circuit_consec()
+        circuit_nonconsec()
+
+        assert np.allclose(dev_consec.state, dev_nonconsec.state)
+
+    @pytest.mark.parametrize("template, diffable, nondiffable, n_wires", CV_DIFFABLE_NONDIFFABLE)
+    def test_cv_result_is_wire_label_independent(self, template, diffable, nondiffable, n_wires, gaussian_dummy):
+        """Tests integration of cv templates with non-integer and non-consecutive wires."""
+
+        # Merge differentiable and non-differentiable arguments
+        nondiffable.update(diffable)
+
+        # Construct qnode with consecutive wires
+        dev_consec = gaussian_dummy(wires=nondiffable['wires'])
+
+        @qml.qnode(dev_consec)
+        def circuit_consec():
+            template(**nondiffable)
+            return qml.expval(qml.Identity(wires=0))
+
+        # Construct qnode with nonconsecutive wires
+        kwargs_nonconsec = nondiffable.copy()
+        nonconsecutive_wires = np.random.choice(['a', 'b', 'c', 'd', 'e', 'f'], size=n_wires, replace=False)
+        kwargs_nonconsec['wires'] = nonconsecutive_wires
+        dev_nonconsec = gaussian_dummy(wires=nonconsecutive_wires)
+
+        @qml.qnode(dev_nonconsec)
+        def circuit_nonconsec():
+            template(**kwargs_nonconsec)
+            return qml.expval(qml.Identity(wires=nonconsecutive_wires[0]))
+
+        circuit_consec()
+        circuit_nonconsec()
+
+        assert np.allclose(dev_consec._state[0], dev_nonconsec._state[0])
```

Comment:
There's a lot going on in these tests that is masking what is actually being tested. Suggest simplifying, even if it means some code duplication

Reply:
You are right - the tests are terrible to read. My hope was to not make new fixtures, because at the moment a template developer can just add a standard template to the list of fixtures and all tests are taken care of. But I know it is hacky! I will rewrite this, even though it may require the entire integration test structure to be rewritten :/